src/main/java/org/apache/sling/graphql/schema/aggregator/impl/Partial.java

```diff
@@ -48,12 +47,24 @@
         TYPES
     }
 
-    /** The name of this partial */
-    @NotNull String getName();
+    /**
+     * Returns the partial info.
+     *
+     * @return the partial info
+     */
+    @NotNull PartialInfo getPartialInfo();
 
     /** Return a specific section of the partial, by name */
     @NotNull Optional<Section> getSection(SectionName name);
 
     /** Names of the Partials on which this one depends */
-    @NotNull Set<String> getRequiredPartialNames();
+    @NotNull Set<PartialInfo> getRequiredPartialNames();
+
+    /**
+     * Returns the SHA256 hash of the source that was used to build this partial.
+     *
+     * @return the SHA256 hash of the source that was used to build this partial
+     */
+    @NotNull String getSHA256Hash();
```

Comment:
Shouldn't the name just be `getDigest`? The algorithm used is an implementation detail IMHO

Reply:
I was initially thinking that two partials should be equal irrespective of where they are coming from and hence wanted to force the various
implementations to use the same hash algorithm.

Reply:
Let's do it like this then, because it's the best compromise: we call the method `getDigest / getHash` but we force implementations to provide the output in this format:
```
algorithm: hash
```

This way two partials can be the same irrespective of their provider if the various implementations hash the source in the same way.

Reply:
Implemented in https://github.com/apache/sling-org-apache-sling-graphql-schema-aggregator/pull/2/commits/7a2bba8c5e9321212e6d5708563c4fe169ebe589.